src/ape/utils.py

```diff
@@ -384,6 +396,32 @@ def get_release(self, repo_path: str, version: str):
 
         return repo.get_release(version)
 
+    def get_repo(self, repo_path: str) -> GithubRepository:
+        if repo_path not in self._repo_cache:
+            self._repo_cache[repo_path] = self._client.get_repo(repo_path)
+
+        return self._repo_cache[repo_path]
+
+    def clone_repo(
+        self, repo_path: str, target_path: Path, branch: Optional[str] = None
+    ) -> GitRepository:
+        """
+        Clone a repository from Github.
+
+        Args:
+            repo_path (str): The path on Github to the repository,
+                            e.g. ``OpenZeppelin/openzeppelin-contracts``.
+            target_path (Path): The local path to store the repo.
+            branch (Optional[str]): The branch to clone. Defaults to the default branch.
+        """
+
+        repo = self.get_repo(repo_path)
+        branch = branch or repo.default_branch
+        logger.info(f"Cloning branch '{branch}' from '{repo.name}'.")
+
+        clone = pygit2.clone_repository(repo.git_url, str(target_path), checkout_branch=branch)
```

Comment:
https://github.com/libgit2/pygit2/issues/1119

^ I tried making a progress bar for cloning but was not successful